### ISSUE TITLE
Error on snippets without key if helm-yas-display-key-on-candidate is non-nil

### ISSUE THREAD

**Issue by Szunti** [emacs-jp/helm-c-yasnippet#17]
An example of a problematic snippet is https://github.com/AndreaCrotti/yasnippet-snippets/blob/master/snippets/makefile-gmake-mode/template:
```
# -*- mode: snippet -*-
# name: template
# contributor: gbalats
# binding: C-c C-t
# --
define ${1:PROGRAM$(upcase yas-text)}_template
$0
endef

\$(foreach ${2:${1:$(downcase yas-text)}},\$(${3:$1S}),\$(eval \$(call $1_template,\$($2))))
```

Backtrace:
```
Debugger entered--Lisp error: (wrong-type-argument stringp [3 20])
  propertize([3 20] face helm-yas-key)
  helm-yas-get-transformed-list(((template-key-alist ("$(wildcard $0)" . "wl"...
  (lambda (candidates) (helm-yas-get-transformed-list helm-yas-cur-snippets-a...
  apply((lambda (candidates) (helm-yas-get-transformed-list helm-yas-cur-snip...
  helm-funcall-with-source(((name . "Yasnippet") (init lambda nil (setq helm-...
  helm-process-candidate-transformer((helm-yas-get-candidates helm-yas-cur-sn...
  helm-transform-candidates((helm-yas-get-candidates helm-yas-cur-snippets-al...
  helm-get-candidates(((name . "Yasnippet") (init lambda nil (setq helm-yas-s...
  helm-get-cached-candidates(((name . "Yasnippet") (init lambda nil (setq hel...
  helm-compute-matches(((name . "Yasnippet") (init lambda nil (setq helm-yas-...
  helm--collect-matches((((name . "Yasnippet") (init lambda nil (setq helm-ya...
  helm-update(nil)
  helm-read-pattern-maybe(nil nil nil nil nil nil nil)
  helm-internal(helm-source-yasnippet)
  apply(helm-internal helm-source-yasnippet)
  helm(helm-source-yasnippet)
  helm-yas-complete()
  funcall-interactively(helm-yas-complete)
  call-interactively(helm-yas-complete)
  spacemacs/helm-yas()
  funcall-interactively(spacemacs/helm-yas)
  call-interactively(spacemacs/helm-yas nil nil)
  command-execute(spacemacs/helm-yas)
```

**Comment by syohex** [comment]
Sorry very rate reply. I have merged #20. Please check latest version

**Comment by Szunti** [comment]
Updated the packages in Spacemacs and it works now. Thanks.
